market-generator/src/test/java/market/MarketServiceImplTest.java

```diff
@@ -0,0 +1,55 @@
+package market;
+
+import com.market.MarketDataRequest;
+import com.market.TickResponse;
+import com.market.banica.generator.service.MarketServiceImpl;
+import com.market.banica.generator.service.MarketSubscriptionManager;
+import com.market.banica.generator.service.TickGenerator;
+import io.grpc.stub.StreamObserver;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class MarketServiceImplTest {
+
+    @InjectMocks
+    private MarketServiceImpl marketService;
+
+    @Mock
+    private MarketSubscriptionManager marketSubscriptionServiceImpl;
+
+    @Mock
+    private TickGenerator tickGenerator;
+
+    private static final String GOOD_BANICA = "Banica";
+
+    private static final MarketDataRequest MARKET_DATA_REQUEST_BANICA = MarketDataRequest.newBuilder().setGoodName(GOOD_BANICA).build();
+
+    private final StreamObserver<TickResponse> subscriberOne = mock(StreamObserver.class);
+
+    @Test
+    void subscribeForItem() {
```

Comment:
Suggestion: Use the following convention for test names: "whatTheTestDoes/WhatParamsItTakes/Result
Example: subscribeForItemWithValidDataInputExcutesWithSuccess